### PR TITLE
CXP-1426: Limit the number of custom apps (created)

### PR DESCRIPTION
**Pull request by eoliveauakeneo**
**Description (for Contributor and Core Developer)**
image

image

### CODE REVIEW

File: src/Akeneo/Connectivity/Connection/back/tests/EndToEnd/CustomApps/Controller/Internal/IsCustomAppsNumberLimitReachedActionEndToEnd.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+        $this->featureFlags = $this->get('feature_flags');
+        $this->customAppLoader = $this->get(CustomAppLoader::class);
+    }
+    protected function getConfiguration(): Configuration
+    {
+        return $this->catalog->useMinimalCatalog();
+    }
+
+    public function test_it_returns_custom_app_limit_reached_flag(): void
```
Comment (guillaumefer-akeneo) [inline src/Akeneo/Connectivity/Connection/back/tests/EndToEnd/CustomApps/Controller/Internal/IsCustomAppsNumberLimitReachedActionEndToEnd.php:35]:
```suggestion
    public function test_it_returns_custom_app_limit_not_reached_flag(): void
```

File: src/Akeneo/Connectivity/Connection/front/src/infrastructure/Marketplace.tsx [hunk 2]
```diff
@@ -4,22 +4,34 @@ import {AkeneoThemeProvider} from './akeneo-theme-provider';
 import {withDependencies} from './dependencies-provider';
 import {SelectUserProfilePage} from '../connect/pages/SelectUserProfilePage';
 import {MarketplacePage} from '../connect/pages/MarketplacePage';
+import {QueryClient, QueryClientProvider} from 'react-query';
+
+const client = new QueryClient({
+    defaultOptions: {
```
Comment (guillaumefer-akeneo) [inline src/Akeneo/Connectivity/Connection/front/src/infrastructure/Marketplace.tsx:10]: why a cache ?

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/CustomApps/Controller/External/CreateCustomAppAction.php [hunk 3]
```diff
@@ -66,7 +66,11 @@ public function __invoke(Request $request): JsonResponse
 
         $errors = [];
         foreach ($this->validator->validate($createCommand) as $violation) {
-            $snakeCasedPropertyPath = \strtolower(\preg_replace("/[A-Z]/", '_$0', $violation->getPropertyPath()));
+            if ($violation->getPropertyPath() == '') {
```
[the remark this answers is not in the document]
Comment (eoliveauakeneo) [reply src/Akeneo/Connectivity/Connection/back/Infrastructure/CustomApps/Controller/External/CreateCustomAppAction.php:69]: should not be here, handled on front side

File: src/Akeneo/Connectivity/Connection/front/src/infrastructure/Marketplace.tsx (hunk 2 above)
Comment (eoliveauakeneo) [reply src/Akeneo/Connectivity/Connection/front/src/infrastructure/Marketplace.tsx:10]: ISO catalog config

**Review by guillaumefer-akeneo — APPROVED** [review]

**Review by RiffVentura — APPROVED** [review]

**Pull request merged**